Experiment automation/(LUMICKS) Reference kit automation script/Catch_beads_DNA_record_Kymo_for_LacI_version_1.py

```diff
@@ -0,0 +1,318 @@
+    while bl.fluidics.pressure < target:
+        bl.fluidics.increase_pressure()
+        bl.pause(1.0)
+
+
+def start_flow(pressure):
+    print("Starting Flow.")
+    set_pressure(pressure)
+    bl.fluidics.open(1, 2, 3, 6)
+
+
+def stop_flow():
+    print("Stopping flow.")
+    bl.fluidics.close(1, 2, 3, 6)
+
+
+def catch_beads(match_threshold, pressure=0.25):
+    """Starts the flow and attempts to catch beads. Toggles the shutters when match score is too low."""
+    start_flow(pressure)
+
+    print(f"Waiting {time_to_wait_for_flow} sec for flow to begin.")
+    bl.pause(time_to_wait_for_flow)
+
+    print("Moving to bead channel.")
+    bl.microstage.move_to(name_bead_channel)
+    bl.pause(1.0)
+    bl.mirror1.clear()
+    bl.mirror2.clear()
+
+    print("Trapping beads.")
+    start_time = time.time()
+    n = 0
+    while match_score1.latest_value < match_threshold or match_score2.latest_value < match_threshold:
+        """Drop beads that do not fulfill the template"""
+        if 0 < match_score1.latest_value < match_threshold:
+            bl.mirror1.clear()
+        if 0 < match_score2.latest_value < match_threshold:
+            bl.mirror2.clear()
+        if n > 50:
+            stop_flow()
+            raise Exception("Need more than 750 seconds to catch beads, stop script")
+        """If it's taking too long, maybe something is stuck in the trap. Clear both traps."""
+        if time.time() - start_time > 15:
+            bl.mirror1.clear()
+            bl.mirror2.clear()
+            start_time = time.time()
+            n += 1
+        bl.pause(1.0)
+
+    print("Got beads!")
+
+
+def goto_distance(target, match_threshold, speed=1, tolerance=0.2):
+    """Move trap 1 until it reaches the `target` distance from trap 2.
+
+    Note: This throws an error if the beads are lost (since we will not have a reliable distance then either)"""
+    dx = target - distance.latest_value
+    throw_if_beads_lost(match_threshold)
+
+    while abs(dx) > tolerance:  # um
+        bl.mirror1.move_by(dx=0.1 if dx > 0 else -0.1, speed=speed)
+        dx = target - distance.latest_value
+        throw_if_beads_lost(match_threshold)
+
+
+def goto_force(target, match_threshold, speed=1, tolerance=1, tether_lost_threshold=5):
+    """Move trap 1 until it reaches the `target` force on trap 2.
+
+    Note: This throws an error if the beads are lost"""
+    df = target - force.latest_value
+    throw_if_beads_lost(match_threshold)
+
+    while abs(df) > tolerance:  # um
+        bl.mirror1.move_by(dx=0.05 if df > 0 else -0.05, speed=speed)
+        df = target - get_force(dt=0.2)
+        throw_if_beads_lost(match_threshold)
+    check_tether_breakage(tether_lost_threshold)
+
+
+def get_force(dt=0.5):
+    t0 = bl.timeline.now
+    bl.pause(dt)
+    t1 = bl.timeline.now
+    f = np.mean(force[t0:t1].data)
+
+    return f
+
+
+def catch_dna(min_distance, max_distance, match_threshold, force_threshold, fishing_speed, fishing_attempts, dt):

```

Comment:
Several of these parameters are not immediately obvious. Would be good to document the parameters of these functions. `fishing_attempts`? `dt`?